sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -402,15 +402,20 @@ static int _mlme_link_adr_req(gnrc_lorawan_t *mac, uint8_t *p, size_t len, uint8
     return consumed_bytes;
 }
 
-int _fopts_mlme_link_adr_ans(lorawan_buffer_t *buf)
+int _fopts_mlme_link_adr_ans(gnrc_lorawan_t *mac, lorawan_buffer_t *buf)
 {
-    if (buf) {
-        assert(buf->index + GNRC_LORAWAN_CID_SIZE <= buf->size);
-        buf->data[buf->index++] = GNRC_LORAWAN_CID_LINK_ADR_ANS;
-        buf->data[buf->index++] = 0x7;
+    uint8_t _count = mac->mlme.adr_req_cnt;
+
+    if (buf){
+        while(_count) {
```

Comment:
I think a for loop is a better choice here

Reply:
Done.